bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -154,19 +228,18 @@ private void manageCurPosIfEnvContainsTarget(final SPTarget target, final Runnab
         final TargetEnvironment env = obsComp.getTargetEnvironment();
         if (env == null || !env.getTargets().contains(target)) return;
 
-        if (_curPos != null)
-            _curPos.deleteWatcher(posWatcher);
+        // If current selection is a target, remove the posWatcher.
+        _curSelection.foreach(e -> e.forEachLeft(t -> t.deleteWatcher(posWatcher)));
```

Comment:
Maybe we should be consistent with scala and use `foreach` instead of `forEach`.